### ISSUE TITLE
Internal build failure "Signing validation failed"

### ISSUE THREAD

**Issue by echesakovMSFT** [dotnet/coreclr#27586]
We haven't had a successful official build in coreclr repo since October 29 due to the failure in **Validate Signing Validation** job

Example of the issue https://dnceng.visualstudio.com/internal/_build/results?buildId=408632
```
  There are unsigned files.
  Total Time: 00:00:34.8090883
  Total Files: 2189, Signed: 380, Unsigned: 14, Skipped: 1573, Excluded: 1245, Skipped & Excluded: 1023
D:\a\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19463.3\tools\SdkTasks\SigningValidation.proj(56,5): error : Signing validation failed. Check signcheck.errors.log for more information.
##[error].packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19463.3\tools\SdkTasks\SigningValidation.proj(56,5): error : Signing validation failed. Check signcheck.errors.log for more information.
```

@dotnet/coreclr-infra

**Comment by elinor-fung** [comment]
Looks like it is failing on the crossgen2 files.
```
 [File] runtime.win-x64.Microsoft.NETCore.Crossgen2.5.0.0-alpha1.19528.4.nupkg, Signed: False
    ...
    [File] 0cb43a47ce1757c6fc92aeccc9630d3a.exe, Signed: False, Full Name: tools/crossgen2.exe [Error] HRESULT: 800b0100 (No signature was present in the subject)
    [File] da06fa01f1fe4f184c20ef57be4b3f3c.dll, Signed: False, Full Name: tools/crossgen2.dll [Error] HRESULT: 800b0100 (No signature was present in the subject)
    [File] 910c3fe803caea000cb1c793733b7cc7.dll, Signed: False, Full Name: tools/ILCompiler.DependencyAnalysisFramework.dll [Error] HRESULT: 800b0100 (No signature was present in the subject)
    [File] b72c11d86e7d368e832d8a90c4ad80fa.dll, Signed: False, Full Name: tools/ILCompiler.ReadyToRun.dll [Error] HRESULT: 800b0100 (No signature was present in the subject)
    [File] 10c9c91798d5d934b79c9f0511619784.dll, Signed: False, Full Name: tools/ILCompiler.TypeSystem.ReadyToRun.dll [Error] HRESULT: 800b0100 (No signature was present in the subject)
    ...
    [File] 82754612958358bca1e26cf459897333.dll, Signed: False, Full Name: tools/clrjitilc.dll [Error] HRESULT: 800b0100 (No signature was present in the subject)
    [File] cc3d128c91c03768cebed0e65dfdab7f.dll, Signed: False, Full Name: tools/jitinterface.dll [Error] HRESULT: 800b0100 (No signature was present in the subject)
```

**Comment by trylek** [comment]
Hmm, that is unfortunate. Does that mean we should start signing Crossgen2 or somehow exempt it from the signing checks? I suspect that long-term it's the former - I can easily follow up on that but I must admit I'm not familiar with the signing process so I'll probably need help with that.

**Comment by elinor-fung** [comment]
We specify `ItemsToSign` in `eng\Signing.props`. Including those files in that item group should get them signed (I think).

If anything should actually be excluded, we have `eng\SignCheckExclusionsFile.txt`

**Comment by jashook** [comment]
I think @sergiy-k  is worth looping into this.

**Comment by trylek** [comment]
Good point especially as I'm leaving shortly so I probably won't have a chance to make the change myself and it's worth prioritizing to unblock the official build. @jashook - how is this change supposed to be tested? You always told me to run the official pipeline in test signing mode and I assume that's going to fail the signing validation even with the fix - is it possible / allowed to run the official pipeline in real signing mode for the purpose of this validation?

**Comment by jashook** [comment]
We can now real sign as the build will not be published to the bar unless it is run off of master.

**Comment by elinor-fung** [comment]
@trylek fixed this with #27598. Latest official build has passed signing validation.
